org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/fix/helper/WhileToForEach.java

```diff
@@ -418,19 +419,17 @@ public void rewrite(UseIteratorToForLoopFixCore upp, final WhileLoopToChangeHit
 			varBinding= type.resolveBinding();
 		}
 		if (type == null || varBinding == null) {
-			looptargettype= "java.lang.Object"; //$NON-NLS-1$
 			ITypeBinding binding= computeTypeArgument(hit.iteratorDeclaration);
-			Type collectionType= null;
+			Type parametrizedType= null;
```

Comment:
Change to parameterizedType.

Reply:
Done. Also corrected the type in the handleParametrizedType().